src/slurm_plugin/slurm_resources.py

```diff
@@ -246,6 +255,13 @@ def get_compute_resource_name(self):
         _, _, compute_resource_name = parse_nodename(self.name)
         return compute_resource_name
 
+    def _parse_error_code(self):
+        """Parse RunInstance error code from node reason."""
+        if not self.reason:
+            return None
+        matches = re.match(r"^\(Code:([a-z|A-Z]+)\)Failure when resuming nodes$", self.reason)
```

Comment:
we might want to compare the regex based approach with the following one to see which one is faster:
1. Check if the sting starts with "(Code:"
2. use find to get the index of `)`
3. Extract the code with string[6:index_of_bracket]

Reply:
Your proposed way is 10 times faster than the regex approach: the second way take about 
--- 2.21729278564e-05 seconds ---
Regex:
--- 0.000202894210815 seconds ---
I switched to the way you proposed